dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs

```diff
@@ -94,100 +95,277 @@ public Plan CreatePlan(string goal)
         return plan;
     }
 
+    /// <summary>
+    /// Execute a plan
+    /// </summary>
+    /// <param name="question">The question to answer</param>
+    /// <param name="context">The context to use</param>
+    /// <param name="token">The cancellation token</param>
+    /// <returns>The context with the result</returns>
+    /// <exception cref="SKException">No AIService available for getting completions.</exception>
     [SKFunction, SKName("ExecutePlan"), Description("Execute a plan")]
     public async Task<SKContext> ExecutePlanAsync(
```

Comment:
Please split the method into smaller ones for better readability and maintenance.

Reply:
Yeah, it's a bit unwieldly.

Reply:
Okay, putting a stake in this one for now.  I've started to go down a few paths for future improvement:

- Define JSON schema for Action and allow Action[] to be returned
- Extract the engine components into their own class to remove coupling with chat history
- Reconsider whether we show invalid results as assistant messages to encourage learning at the trade off of keeping it aligned to expectations.
- Significant test automation to reduce manual burdens.

Reply:
cc @hario90